web/src/app/validation/ceg/duplicate-io-variable-validator.ts

```diff
@@ -7,33 +7,49 @@ import { ValidationMessage } from '../validation-message';
 import { ValidationResult } from '../validation-result';
 import { Validator } from '../validator-decorator';
 
+type IO = ('input' | 'output');
+
 @Validator(CEGModel)
 export class DuplicateIOVariableValidator extends ElementValidatorBase<CEGModel> {
     public validate(element: CEGModel, contents: IContainer[]): ValidationResult {
-        let variableMap: { [variable: string]: string } = {};
-        const invalidNodes: IContainer[] = [];
+
+        const nodeMap: { [variable: string]: IContainer[] } = {};
+        const typeMap: { [variable: string]: IO[] } = {};
+        let invalidNodes: IContainer[] = [];
+
         for (let content of contents) {
             if (!Type.is(content, CEGNode)) {
                 continue;
             }
-            let node: CEGNode = content as CEGNode;
-            let type: string;
+
+            const node: CEGNode = content as CEGNode;
+
+            let type: IO;
             if (!node.incomingConnections || node.incomingConnections.length <= 0) {
                 type = 'input';
             } else if (!node.outgoingConnections || node.outgoingConnections.length <= 0) {
                 type = 'output';
-            } else {
-                type = 'intermediate';
             }
-            let existing: string = variableMap[node.variable];
-            if (existing) {
-                if (existing === 'input' && type === 'output' || existing === 'output' && type === 'input') {
-                    invalidNodes.push(node);
-                }
-            } else {
-                variableMap[node.variable] = type;
+
+            if (typeMap[node.variable.trim()] === undefined) {
+                typeMap[node.variable.trim()] = [];
+            }
+            if (!typeMap[node.variable.trim()].includes(type)) {

```

Comment:
This line leads to a TS-compile-error on Travis. We should use this corrected version (I already updated this on develop).

```suggestion
            if (typeMap[node.variable.trim()].indexOf(type) < 0) {

```